File: ruby/space-age/space_age.rb
# Write your code for the 'Space Age' exercise in this file. Make the tests in
# `space_age_test.rb` pass.
#
# To get started with TDD, see the `README.md` file in your
# `ruby/space-age` directory.

class SpaceAge
  def initialize(seconds)
    @seconds = seconds
  end

  {
    earth: 1.0,
    mercury: 0.2408467,
    venus: 0.61519726,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132
  }.each_pair do |planet, ratio|
    define_method("on_#{planet}") do
      @seconds / (EARCH_YEAR_IN_SECONDS * ratio)
    end
  end

  private attr_reader :seconds

  EARCH_YEAR_IN_SECONDS = 31_557_600
  private_constant :EARCH_YEAR_IN_SECONDS
end

File: go/space-age/space_age.go
package space

type Planet string

func Age(seconds float64, planet Planet) float64 {
	switch planet {
	case "Earth":
		return seconds / 31557600
	case "Mercury":
		return seconds / 31557600 / 0.240846
	case "Venus":
		return seconds / 31557600 / 0.61519726
	case "Mars":
		return seconds / 31557600 / 1.8808158
	case "Jupiter":
		return seconds / 31557600 / 11.862615
	case "Saturn":
		return seconds / 31557600 / 29.447498
	case "Uranus":
		return seconds / 31557600 / 84.016846
	case "Neptune":
		return seconds / 31557600 / 164.79132
	default:
		return -1
	}
}
